### ISSUE TITLE
Segment_sum

### ISSUE THREAD

**Issue by swamidass** [fferflo/einx#23]
Is there a way to do the equivalent of segment_sum (and product, softmax, etc) in einx?

https://docs.jax.dev/en/latest/_autosummary/jax.ops.segment_sum.html jax.ops.segment_sum — JAX documentation
